app_dart/lib/src/model/firestore/task.dart

```diff
@@ -161,25 +163,28 @@ class Task extends Document {
       log.warning('Tags: $tags');
       throw const BadRequestException('build_address does not contain build number');
     }
-    fields!['buildNumber'] = Value(integerValue: buildAddress.split('/').last);
-    fields!['createTimestamp'] = Value(integerValue: (build.createdTimestamp?.millisecondsSinceEpoch ?? 0).toString());
-    fields!['startTimestamp'] = Value(integerValue: (build.startedTimestamp?.millisecondsSinceEpoch ?? 0).toString());
-    fields!['endTimestamp'] = Value(integerValue: (build.completedTimestamp?.millisecondsSinceEpoch ?? 0).toString());
+    fields![kTaskBuildNumberField] = Value(integerValue: buildAddress.split('/').last);
+    fields![kTaskCreateTimestampField] =
+        Value(integerValue: (build.createdTimestamp?.millisecondsSinceEpoch ?? 0).toString());
+    fields![kTaskStartTimestampField] =
+        Value(integerValue: (build.startedTimestamp?.millisecondsSinceEpoch ?? 0).toString());
+    fields![kTaskEndTimestampField] =
+        Value(integerValue: (build.completedTimestamp?.millisecondsSinceEpoch ?? 0).toString());
 
     _setStatusFromLuciStatus(build);
   }
 
   void resetAsRetry({int attempt = 1}) {
-    name = '$kDatabase/documents/tasks/${commitSha}_${taskName}_$attempt';
+    name = '$kDatabase/documents/$kTaskCollectionId/${commitSha}_${taskName}_$attempt';
     fields = <String, Value>{
-      'createTimestamp': Value(integerValue: DateTime.now().millisecondsSinceEpoch.toString()),
-      'endTimestamp': Value(integerValue: '0'),
-      'bringup': Value(booleanValue: bringup),
-      'name': Value(stringValue: taskName),
-      'startTimestamp': Value(integerValue: '0'),
-      'status': Value(stringValue: Task.statusNew),
-      'testFlaky': Value(booleanValue: false),
-      'commitSha': Value(stringValue: commitSha),
+      kTaskCreateTimestampField: Value(integerValue: DateTime.now().millisecondsSinceEpoch.toString()),
+      kTaskEndTimestampField: Value(integerValue: '0'),
```

Comment:
Can this simply be 0 instead of '0'?

Reply:
I think that needs a string

Reply:
The `integerValue` is confusingly a `String` type: https://github.com/google/googleapis.dart/blob/master/generated/googleapis/lib/firestore/v1.dart#L5732